pkg/service/service.go

```diff
@@ -160,96 +153,45 @@ func (s *Service) Run() error {
 	}
 }
 
-func (s *Service) isIdle() bool {
-	idle := true
-	s.processes.Range(func(key, value interface{}) bool {
-		idle = false
-		return false
+func (s *Service) getIngressInfo(ctx context.Context, req *livekit.GetIngressInfoRequest) (int, *livekit.GetIngressInfoResponse, error) {
+	race := rpc.NewRace[livekit.GetIngressInfoResponse](ctx)
+	race.Go(func(ctx context.Context) (*livekit.GetIngressInfoResponse, error) {
+		return s.rpcServer.SendGetIngressInfoRequest(ctx, req)
+	})
+	race.Go(func(ctx context.Context) (*livekit.GetIngressInfoResponse, error) {
+		return s.psrpcClient.GetIngressInfo(ctx, req)
 	})
-	return idle
+	return race.Wait()
 }
 
-func (s *Service) sendUpdate(ctx context.Context, info *livekit.IngressInfo, err error) {
+func (s *Service) sendUpdate(ctx context.Context, info *livekit.IngressInfo, version int, err error) {
 	if err != nil {
 		info.State.Status = livekit.IngressState_ENDPOINT_ERROR
 		info.State.Error = err.Error()
 		logger.Errorw("ingress failed", errors.New(info.State.Error))
 	}
 
-	if err := s.rpcServer.SendUpdate(ctx, info.IngressId, info.State); err != nil {
-		logger.Errorw("failed to send update", err)
+	if version == 0 {
+		if err := s.rpcServer.SendUpdate(ctx, info.IngressId, info.State); err != nil {
+			logger.Errorw("failed to send update", err)
+		}
+	} else {
+		_, err = s.psrpcClient.UpdateIngressState(ctx, &livekit.UpdateIngressStateRequest{
+			IngressId: info.IngressId,
+			State:     info.State,
+		})
+		if err != nil {
+			logger.Errorw("failed to send update", err)
+		}
 	}
 }
 
-func (s *Service) launchHandler(ctx context.Context, resp *livekit.GetIngressInfoResponse) {
-	// TODO send update on failure
-	ctx, span := tracer.Start(ctx, "Service.launchHandler")
-	defer span.End()
-
-	confString, err := yaml.Marshal(s.conf)
-	if err != nil {
-		span.RecordError(err)
-		logger.Errorw("could not marshal config", err)
-		return
-	}
-
-	infoString, err := protojson.Marshal(resp.Info)
-	if err != nil {
-		span.RecordError(err)
-		logger.Errorw("could not marshal request", err)
-		return
-	}
-
-	args := []string{
-		"run-handler",
-		"--config-body", string(confString),
-		"--info", string(infoString),
-	}
-
-	if resp.WsUrl != "" {
-		args = append(args, "--ws-url", resp.WsUrl)
-	}
-	if resp.Token != "" {
-		args = append(args, "--token", resp.Token)
-	}
-
-	cmd := exec.Command("ingress",
-		args...,
-	)
-
-	cmd.Dir = "/"
-	cmd.Stdout = os.Stdout
-	cmd.Stderr = os.Stderr
-
-	s.monitor.IngressStarted(resp.Info)
-	s.processes.Store(resp.Info.IngressId, &process{
-		info: resp.Info,
-		cmd:  cmd,
-	})
-	defer func() {
-		s.monitor.IngressEnded(resp.Info)
-		s.processes.Delete(resp.Info.IngressId)
-	}()
-
-	err = cmd.Run()
-	if err != nil {
-		logger.Errorw("could not launch handler", err)
-	}
+func (s *Service) Status() ([]byte, error) {
```

Comment:
I believe this is not used anymore after you (rightfully) refactored the CanAccept logic. 

It is possible we may use it again once we add a debug handler to ingress, as we did for egress, but even then we're likely to need a slightly different output. Should we just delete this method? As well as the ProcessManager status method potentially?